PhysicsTools/PatUtils/plugins/ShiftedPFCandidateProducerForPFNoPUMEt.cc

```diff
@@ -72,12 +72,12 @@ void ShiftedPFCandidateProducerForPFNoPUMEt::produce(edm::Event& evt, const edm:
        originalPFCandidate != originalPFCandidates->end();
        ++originalPFCandidate) {
     const reco::PFJet* jet_matched = nullptr;
-    for (std::vector<const reco::PFJet*>::iterator jet = selectedJets.begin(); jet != selectedJets.end(); ++jet) {
-      for (std::vector<reco::PFCandidatePtr>::const_iterator jetConstituent = (*jet)->getPFConstituents().begin();
-           jetConstituent != (*jet)->getPFConstituents().end() && jet_matched == nullptr;
-           ++jetConstituent) {
-        if (deltaR2(originalPFCandidate->p4(), (*jetConstituent)->p4()) < dR2Match)
-          jet_matched = (*jet);
+    for (auto jet : selectedJets) {
+      for (auto jetc : jet->getPFConstituents()) {
+        if (jet_matched)
```

Comment:
It could be even simpler if you ```break``` inside an if block started with L79 below

On the other hand, there could be some difference between the fixed code and the original one if there are two PFCandidates that are within ```dr2Match```. I understand that this is rather unlikely, and if there are two such candidates nobody can say that taking the first one (as with your fix) is worst than taking the last one (as in the original code). Therefore I'd be tempted to stick on your proposed faster solution.

Reply:
@perrotta , looks like the original code is also taking the first match (due to ` && jet_matched == nullptr` condition in the inner loop) .  @mrodozov I think something like the following will simplify and improve the code

```
    for (auto jet : selectedJets) {
      for (auto jetc : jet->getPFConstituents()) {
        if (deltaR2(originalPFCandidate->p4(), jetc->p4()) < dR2Match) {
          jet_matched = jet;
          break;
        }
      }
      if (jet_matched)
        break;
    }
```

there is no need to loop over all jets if match is already found ... what do you think @perrotta ?

Reply:
Correct.
Yes, that's even better: please @mrodozov  go ahead with it